app/models/ont/request_factory.rb

```diff
@@ -8,73 +8,59 @@ module Ont
   class RequestFactory
     include ActiveModel::Model
 
-    validate :check_request, :check_tag, :check_well_request_join
-    # Note that @ont_request and @sample are validated by association with @request
+    validate :check_ont_request, :check_tag_taggable, :check_well_request_join
 
     def initialize(attributes = {})
       @well = attributes[:well]
-      @tag_service = attributes[:tag_service]
       return unless attributes.key?(:request_attributes)
 
       build_request(attributes[:request_attributes])
     end
 
-    attr_reader :request
+    attr_reader :ont_request
 
     def save(**options)
       return false unless options[:validate] == false || valid?
 
       # No need to validate any lower level objects since validation above has already checked them
-      request.save(validate: false)
-      tag_taggable&.save(validate: false)
+      ont_request.save(validate: false)
```

Comment:
Can we now wrap this in a transaction as each object is singular?

Reply:
Do you mean something like:
```
ActiveRecord::Base.transaction do
  ont_request.save(validate: false)
  tag_taggable.save(validate: false)
  well_request_join.save(validate: false)
end
```

Reply:
What benefit does this have? Should we be wrapping all factory saves in a transaction like this?

Reply:
It would not benefit in terms of performance, but it would (assuming it works) ensure that if any save fails, the data gets rolled back.  We aren't asking for validation on each save, so I suspect the save would only fail if the database fails to respond to our request.

Reply:
Yes. Exactly what I meant.